src/test/java/org/trebol/operation/services/SalesProcessServiceImplTest.java

```diff
@@ -0,0 +1,454 @@
+package org.trebol.operation.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+import static org.trebol.config.Constants.SELL_STATUS_PAYMENT_STARTED;
+import static org.trebol.config.Constants.SELL_STATUS_PENDING;
+import static org.trebol.config.Constants.SELL_STATUS_PAYMENT_CANCELLED;
+import static org.trebol.config.Constants.SELL_STATUS_PAYMENT_FAILED;
+import static org.trebol.config.Constants.SELL_STATUS_PAID_UNCONFIRMED;
+import static org.trebol.config.Constants.SELL_STATUS_PAID_CONFIRMED;
+import static org.trebol.config.Constants.SELL_STATUS_REJECTED;
+import static org.trebol.config.Constants.SELL_STATUS_COMPLETED;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.trebol.exceptions.BadInputException;
+import org.trebol.jpa.entities.Product;
+import org.trebol.jpa.entities.Sell;
+import org.trebol.jpa.entities.SellStatus;
+import org.trebol.jpa.repositories.ISalesJpaRepository;
+import org.trebol.jpa.repositories.ISellDetailsJpaRepository;
+import org.trebol.jpa.repositories.ISellStatusesJpaRepository;
+import org.trebol.jpa.services.GenericCrudJpaService;
+import org.trebol.jpa.services.ITwoWayConverterJpaService;
+import org.trebol.pojo.ProductPojo;
+import org.trebol.pojo.SellPojo;
+
+@ExtendWith(MockitoExtension.class)
+class SalesProcessServiceImplTest {
+	
+	@InjectMocks	SalesProcessServiceImpl instance;
```

Comment:
//	@InjectMocks
  SalesProcessServiceImpl instance;
	@Mock	GenericCrudJpaService<SellPojo, Sell> crudService;
	@Mock	ISalesJpaRepository salesRepository;
	ITwoWayConverterJpaService<SellPojo, Sell> converterService;
	ITwoWayConverterJpaService<ProductPojo, Product> productConverterService;
	@Mock	ISellStatusesJpaRepository sellStatusesRepository;
	
	@Mock	ISellDetailsJpaRepository sellDetailsRepository;	
	//@Mock	ProductsConverterJpaServiceImpl productConverterService;

  @BeforeEach
  public void setUpBeforeEach() {
    converterService = (ITwoWayConverterJpaService<SellPojo, Sell>) Mockito.mock(ITwoWayConverterJpaService.class);
    productConverterService = (ITwoWayConverterJpaService<ProductPojo, Product>) Mockito.mock(ITwoWayConverterJpaService.class);
    instance = new SalesProcessServiceImpl(crudService, salesRepository, sellDetailsRepository, sellStatusesRepository, converterService, productConverterService);
  }

We manual inject the dependencies. That's the solution in handling generics

Reply:
yeah. Its kind of tricky but casting the generic dependencies solves the issue

Reply:
No worries man. Salud!